pkg/azclient/utils/tracing.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/tracing"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	semconv "go.opentelemetry.io/otel/semconv/v1.18.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	instrumentationName    = "otlp"
+	instrumentationVersion = "v0.1.0"
+	SDKSpanKindKey         = "azsdkTraceKind"
+)
+
+var (
+	tracer = otel.GetTracerProvider().Tracer(
+		instrumentationName,
+		trace.WithInstrumentationVersion(instrumentationVersion),
+		trace.WithSchemaURL(semconv.SchemaURL),
+	)
+)
+
+func NewOtlpSpan(ctx context.Context, spanName string, options *tracing.SpanOptions) (context.Context, tracing.Span) {
+	var span trace.Span
+	var traceAttributes []attribute.KeyValue
+
+	if options != nil {
+		for _, entry := range options.Attributes {
+			traceAttributes = append(traceAttributes, attribute.String(entry.Key, fmt.Sprintf("%v", entry.Value)))
+		}
+		traceAttributes = append(traceAttributes, attribute.Int(SDKSpanKindKey, int(options.Kind)))
+	}
+
+	ctx, span = tracer.Start(ctx, spanName, trace.WithAttributes(traceAttributes...))
+
+	return ctx, tracing.NewSpan(
+		tracing.SpanImpl{
+			End: func() {
+				span.End()
+			},
+			SetAttributes: func(attributes ...tracing.Attribute) {
+				var traceAttributes []attribute.KeyValue
+				for _, entry := range attributes {
+					traceAttributes = append(traceAttributes, attribute.String(entry.Key, fmt.Sprintf("%v", entry.Value)))
```

Comment:
Is it possible to avoid using %v in our code?

Reply:
attribute. Value is type of interface{}. It will throw panic if the object doesn't implement String() func. I agree we should avoid using %v, but here we don't have better option.